lib/msal-browser/src/controllers/NestedAppAuthController.ts

```diff
@@ -458,20 +469,14 @@ export class NestedAppAuthController implements IController {
         if (!cachedAccessToken) {
             this.logger.verbose("No cached access token found");
             return Promise.resolve(null);
-            // If access token has expired, remove the token from cache and return null
         } else if (
             TimeUtils.wasClockTurnedBack(cachedAccessToken.cachedAt) ||
             TimeUtils.isTokenExpired(
                 cachedAccessToken.expiresOn,
                 this.config.system.tokenRenewalOffsetSeconds
             )
         ) {
-            this.logger.verbose(
-                "Cached access token has expired, deleting all related tokens from cache"
-            );
-            const accountEntity =
-                AccountEntity.createFromAccountInfo(currentAccount);
-            await this.browserStorage.removeAccountContext(accountEntity);
```

Comment:
Reason this was removed?

Reply:
Because access tokens can be generated at various times and we seem to remove all of them if one is `expired`. There should be a better cleaning API for only removing specific tokens IMHO, which I can handle in a separate PR.

Reply:
Oh gotcha, thanks - I believe there is a different API to just remove the expired token, suggest looking at the non-NAA implementation (not a requirement for this PR)